compiler/aarch64/codegen/ControlFlowEvaluator.cpp

```diff
@@ -776,3 +776,27 @@ static bool virtualGuardHelper(TR::Node *node, TR::CodeGenerator *cg)
    return false;
 #endif
    }
+
+TR::Register *OMR::ARM64::TreeEvaluator::igotoEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR_ASSERT(node->getNumChildren()>=1, "at least one child expected for igoto");
+
+   TR::Node *labelAddr = node->getFirstChild();
+   TR::Register *addrReg = cg->evaluate(labelAddr);
+   TR::RegisterDependencyConditions *deps = NULL;
+   if (node->getNumChildren() > 1)
+      {
+      TR_ASSERT(node->getNumChildren() == 2 && node->getChild(1)->getOpCodeValue() == TR::GlRegDeps, "igoto has maximum of two children and second one must be global register dependency");
+      TR::Node *glregdep = node->getChild(1);
+      cg->evaluate(glregdep);
+      deps = generateRegisterDependencyConditions(cg, glregdep, 0);
```

Comment:
This will terminate with a `TR_UNIMPLEMENTED` assert, won't it?

https://github.com/eclipse/omr/blob/9ad00ba7d990723db2c00c6fc30d5485f7471334/compiler/aarch64/codegen/OMRRegisterDependency.cpp#L36

Reply:
I wasn't aware of that. Yes, I think that it will hit `TR_UNIMPLEMENTED` when GRA is enabled.
I found several evaluators calling this `generateRegisterDependencyConditions`. For example:
https://github.com/eclipse/omr/blob/6e4fcab33e5526fd1f0de79d13f5443e9ecc7f9c/compiler/aarch64/codegen/OMRTreeEvaluator.cpp#L826
I feel that my commit is consistent with other evaluators. But if you think it's not, maybe I would remove the if block at line 787-794 or add `TR_UNIMPLEMENTED` inside the if block.

Reply:
As we discussed privately yesterday, we will leave the code as you have it.  I don't want you to remove code that we will need eventually for GlRegDep support, so hitting the eventual assert in the register dependency code is fine as we'll need to implement that eventually when we actually do the GlRegDep support.